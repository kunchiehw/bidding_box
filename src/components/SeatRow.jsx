import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Grid, Segment } from 'semantic-ui-react';
import { SEATS, VULNERABILITIES } from '../utils/constants';
import { isVulnerable } from '../utils/helpers';

const propTypes = {
  vulnerability: PropTypes.oneOf(VULNERABILITIES).isRequired,
};

class SeatRow extends PureComponent {
  render() {
    return (
      <Grid.Row key="seat" className="seat-row" column={4}>
        {SEATS.map((seat) => {
          const seatIsVulnerable = isVulnerable(seat, this.props.vulnerability);
          return ((
            <Grid.Column key={seat}>
              <Segment
                className="seat-cell"
                color={(seatIsVulnerable) ? 'red' : null}
                inverted={seatIsVulnerable}
              >
                {seat}
              </Segment>
            </Grid.Column>
          ));
        })}
      </Grid.Row>
    );
  }
}

SeatRow.propTypes = propTypes;

export default SeatRow;
