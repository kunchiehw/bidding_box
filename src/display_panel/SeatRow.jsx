import React from 'react';
import PropTypes from 'prop-types';
import {SEATS, VULS} from '../util/Util.jsx';

const propTypes = {
  vulnerability: PropTypes.oneOf(VULS).isRequired
};

function SeatBlock(seat, vulnerability) {
  const vulnerable = (vulnerability === 'BOTH')
    || (vulnerability === 'NS' && (seat === 'NORTH' || seat === 'SOUTH'))
    || (vulnerability === 'EW' && (seat === 'EAST' || seat === 'WEST'))

  if(vulnerable) {
    return (
      <div key={seat} className='display-block seat-block vulnerable-seat-block'>{seat}</div>
    );
  }
  else {
    return (
      <div key={seat} className='display-block seat-block non-vulnerable-seat-block'>{seat}</div>
    );
  }
}

function SeatRow(props) {
  const SeatBlocks = [];
  for (var seatIndex = 0; seatIndex < SEATS.length; seatIndex++) {
    SeatBlocks.push(SeatBlock(SEATS[seatIndex], props.vulnerability));
  }
  return (
    <div className='seat-row'>
      {SeatBlocks}
    </div>
  );
}

SeatRow.propTypes = propTypes;

export default SeatRow;
