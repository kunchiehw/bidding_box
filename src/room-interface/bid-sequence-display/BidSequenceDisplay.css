.bid-sequence-display {
  display: inline-block;
  vertical-align: top;
  text-align: left;
  margin: 0px 15px;
}

.bid-sequence-display .bid-sequence-display-cell {
  display: inline-block;
  text-align: center;
  width: 75px;
  min-width: 75px;
  margin: 1px 2px !important;
  padding: .5em 1em;
}

.bid-sequence-display .seat-row .seat-cell {
  border: 1px;
  border-style: solid;
}

.bid-sequence-display .bid-cell.empty-cell {
  visibility: hidden;
}

.bid-sequence-display .bid-cell.PASS-cell {
  border: 2px solid #21ba45;
  color: #21ba45;
}

.bid-sequence-display .bid-cell.DOUBLE-cell {
  border: 2px solid #db2828;
  color: #db2828;
}

.bid-sequence-display .bid-cell.REDOUBLE-cell {
  border: 2px solid #2185d0;
  color: #2185d0;
}
