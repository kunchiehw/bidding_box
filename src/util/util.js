export const SUITS = ['CLUBS', 'DIAMONDS', 'HEARTS', 'SPADES', 'NOTRUMPS'];

export const NON_SUITS = ['PASS', 'DOUBLE', 'REDOUBLE'];

export const SEATS = ['WEST', 'NORTH', 'EAST', 'SOUTH'];

export const SEATS_NS = ['NORTH', 'SOUTH'];

export const SEATS_EW = ['EAST', 'WEST'];

export const PARTICIPANTS_ROLE = ['EAST', 'WEST', 'OBSERVER', 'ALL_SEATS'];

export const VULS = ['NONE', 'NS', 'EW', 'BOTH'];
