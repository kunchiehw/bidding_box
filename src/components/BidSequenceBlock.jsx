import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Divider } from 'semantic-ui-react';
import SeatRow from './SeatRow';
import BidSequenceRows from './BidSequenceRows';
import { SEATS, VULNERABILITIES, ALL_SUITS } from '../utils/constants';

const propTypes = {
  dealer: PropTypes.oneOf(SEATS).isRequired,
  vulnerability: PropTypes.oneOf(VULNERABILITIES).isRequired,
  bidSeq: PropTypes.arrayOf(PropTypes.shape({
    level: PropTypes.number,
    suit: PropTypes.oneOf(ALL_SUITS).isRequired,
  })).isRequired,
};

function BidSequenceBlock(props) {
  return (
    <Grid className="bid-sequence-block">
      <SeatRow vulnerability={props.vulnerability} />
      <Divider />
      <BidSequenceRows dealer={props.dealer} bidSeq={props.bidSeq} />
    </Grid>
  );
}

BidSequenceBlock.propTypes = propTypes;

export default BidSequenceBlock;
