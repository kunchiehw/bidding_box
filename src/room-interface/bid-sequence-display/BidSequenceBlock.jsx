import React from 'react';
import PropTypes from 'prop-types';
import { Segment } from 'semantic-ui-react';
import { bidToString, suitColor } from '../helper-RoomInterface';
import { SEATS, STANDARD_SUITS, ALL_SUITS } from '../../util/util';

const propTypes = {
  dealer: PropTypes.oneOf(SEATS).isRequired,
  bidSeq: PropTypes.arrayOf(PropTypes.shape({
    level: PropTypes.number,
    suit: PropTypes.oneOf(ALL_SUITS).isRequired,
  })).isRequired,
};

function BidSequenceBlock(props) {
  const bidSequenceRows = [];
  let bidSequenceRow = [];
  let bidSequenceRowIndex = 0;

  // Generate empty cells first when the dealer is not WEST (the left most player).
  for (let seatIndex = 0; seatIndex < SEATS.length; seatIndex += 1) {
    if (props.dealer !== SEATS[seatIndex]) {
      bidSequenceRow.push(<Segment key={SEATS[seatIndex]} className="bid-sequence-cell empty-cell"> Empty </Segment>);
    } else {
      break;
    }
  }

  for (let bidSeqIndex = 0; bidSeqIndex < props.bidSeq.length; bidSeqIndex += 1) {
    const bid = props.bidSeq[bidSeqIndex];
    if (STANDARD_SUITS.includes(bid.suit)) {
      bidSequenceRow.push(<Segment key={bidSeqIndex} className="bid-sequence-cell" color={suitColor(bid.suit)} inverted> {bidToString(bid.level, bid.suit)} </Segment>);
    } else {
      bidSequenceRow.push(<Segment key={bidSeqIndex} className={`bid-sequence-cell ${bid.suit}-cell`}> {bidToString(bid.level, bid.suit)} </Segment>);
    }
    if (bidSequenceRow.length === 4) {
      bidSequenceRows.push(<div key={bidSequenceRowIndex} className="bid-sequence-row"> {bidSequenceRow} </div>);
      bidSequenceRow = [];
      bidSequenceRowIndex += 1;
    }
  }

  bidSequenceRows.push(<div key={bidSequenceRowIndex} className="bid-sequence-row"> {bidSequenceRow} </div>);

  return (
    <div className="bid-sequence-block">
      {bidSequenceRows}
    </div>
  );
}

BidSequenceBlock.propTypes = propTypes;

export default BidSequenceBlock;
