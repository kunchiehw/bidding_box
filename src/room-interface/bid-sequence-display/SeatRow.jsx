import React from 'react';
import PropTypes from 'prop-types';
import { SEATS, SEATS_NS, SEATS_EW, VULS } from '../../util/util';

const propTypes = {
  vulnerability: PropTypes.oneOf(VULS).isRequired,
};

function SeatCell(seat, vulnerability) {
  const vulnerable = (vulnerability === 'BOTH')
    || (vulnerability === 'NS' && SEATS_NS.includes(seat))
    || (vulnerability === 'EW' && SEATS_EW.includes(seat));

  if (vulnerable) {
    return (
      <div key={seat} className="bid-sequence-display-cell seat-cell vulnerable-seat-cell">{seat}</div>
    );
  }

  return (
    <div key={seat} className="bid-sequence-display-cell seat-cell non-vulnerable-seat-cell">{seat}</div>
  );
}

function SeatRow(props) {
  const SeatCells = [];
  for (let seatIndex = 0; seatIndex < SEATS.length; seatIndex += 1) {
    SeatCells.push(SeatCell(SEATS[seatIndex], props.vulnerability));
  }
  return (
    <div className="seat-row">
      {SeatCells}
    </div>
  );
}

SeatRow.propTypes = propTypes;

export default SeatRow;
