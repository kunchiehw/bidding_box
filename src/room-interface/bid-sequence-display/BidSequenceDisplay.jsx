import React from 'react';
import PropTypes from 'prop-types';
import { Divider } from 'semantic-ui-react';
import './BidSequenceDisplay.css';
import SeatRow from '../../components/SeatRow';
import BidSequenceBlock from './BidSequenceBlock';
import { SEATS, VULS, ALL_SUITS } from '../../util/util';

const propTypes = {
  dealer: PropTypes.oneOf(SEATS).isRequired,
  vulnerability: PropTypes.oneOf(VULS).isRequired,
  bidSeq: PropTypes.arrayOf(PropTypes.shape({
    level: PropTypes.number,
    suit: PropTypes.oneOf(ALL_SUITS).isRequired,
  })).isRequired,
};

function BidSequenceDisplay(props) {
  return (
    <div className="bid-sequence-display">
      <SeatRow vulnerability={props.vulnerability} />
      <Divider />
      <BidSequenceBlock dealer={props.dealer} bidSeq={props.bidSeq} />
    </div>
  );
}

BidSequenceDisplay.propTypes = propTypes;

export default BidSequenceDisplay;
