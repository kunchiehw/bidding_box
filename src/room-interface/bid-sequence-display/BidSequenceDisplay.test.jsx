import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SeatRow from './SeatRow';
import BidSequenceBlock from './BidSequenceBlock';
import BidSequenceDisplay from './BidSequenceDisplay';
import { SEATS, VULS } from '../../util/util';

describe('SeatRow render', () => {
  VULS.forEach((vul) => {
    it(`vulnerability: ${vul}`, () => {
      const seatRow = shallow(<SeatRow
        vulnerability={vul}
      />);
      expect(toJson(seatRow)).toMatchSnapshot();
    });
  });
});

describe('BidSequenceBlock render', () => {
  const bidSeqs = [
    [],
    [
      { level: 1, suit: 'CLUBS' },
      { level: 2, suit: 'DIAMONDS' },
      { level: 3, suit: 'HEARTS' },
      { level: 4, suit: 'SPADES' },
      { level: 5, suit: 'NOTRUMPS' },
    ],
    [
      { level: 6, suit: 'CLUBS' },
      { level: 0, suit: 'DOUBLE' },
      { level: 0, suit: 'REDOUBLE' },
      { level: 0, suit: 'PASS' },
      { level: 0, suit: 'PASS' },
      { level: 0, suit: 'PASS' },
    ],
  ];

  SEATS.forEach((seat) => {
    bidSeqs.forEach((bidSeq) => {
      it(`seat: ${seat}, bidSeq: ${bidSeq}`, () => {
        const bidSequenceBlock = shallow(<BidSequenceBlock
          dealer={seat}
          bidSeq={bidSeq}
        />);
        expect(toJson(bidSequenceBlock)).toMatchSnapshot();
      });
    });
  });
});

describe('BidSequenceDisplay render', () => {
  it('basic case', () => {
    const bidSequenceDisplay1 = shallow(<BidSequenceDisplay
      dealer="NORTH"
      vulnerability="EW"
      bidSeq={[]}
    />);
    expect(toJson(bidSequenceDisplay1)).toMatchSnapshot();
  });
});
