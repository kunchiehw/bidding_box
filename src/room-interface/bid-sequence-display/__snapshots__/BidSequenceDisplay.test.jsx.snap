// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidSequenceBlock render seat: EAST, bidSeq:  1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell empty-cell"
      key="WEST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell empty-cell"
      key="NORTH"
    >
       Empty 
    </Segment>
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: EAST, bidSeq: [object Object],[object Object],[object Object],[object Object],[object Object] 1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell empty-cell"
      key="WEST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell empty-cell"
      key="NORTH"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="green"
      inverted={true}
      key="0"
    >
       
      1 ♣
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="orange"
      inverted={true}
      key="1"
    >
       
      2 ♦
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="1"
  >
     
    <Segment
      className="bid-sequence-cell"
      color="red"
      inverted={true}
      key="2"
    >
       
      3 ♥
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="blue"
      inverted={true}
      key="3"
    >
       
      4 ♠
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="black"
      inverted={true}
      key="4"
    >
       
      5 N
       
    </Segment>
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: EAST, bidSeq: [object Object],[object Object],[object Object],[object Object],[object Object],[object Object] 1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell empty-cell"
      key="WEST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell empty-cell"
      key="NORTH"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="green"
      inverted={true}
      key="0"
    >
       
      6 ♣
       
    </Segment>
    <Segment
      className="bid-sequence-cell DOUBLE-cell"
      key="1"
    >
       
      X
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="1"
  >
     
    <Segment
      className="bid-sequence-cell REDOUBLE-cell"
      key="2"
    >
       
      XX
       
    </Segment>
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="3"
    >
       
      Pass
       
    </Segment>
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="4"
    >
       
      Pass
       
    </Segment>
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="5"
    >
       
      Pass
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="2"
  >
     
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: NORTH, bidSeq:  1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell empty-cell"
      key="WEST"
    >
       Empty 
    </Segment>
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: NORTH, bidSeq: [object Object],[object Object],[object Object],[object Object],[object Object] 1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell empty-cell"
      key="WEST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="green"
      inverted={true}
      key="0"
    >
       
      1 ♣
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="orange"
      inverted={true}
      key="1"
    >
       
      2 ♦
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="red"
      inverted={true}
      key="2"
    >
       
      3 ♥
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="1"
  >
     
    <Segment
      className="bid-sequence-cell"
      color="blue"
      inverted={true}
      key="3"
    >
       
      4 ♠
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="black"
      inverted={true}
      key="4"
    >
       
      5 N
       
    </Segment>
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: NORTH, bidSeq: [object Object],[object Object],[object Object],[object Object],[object Object],[object Object] 1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell empty-cell"
      key="WEST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="green"
      inverted={true}
      key="0"
    >
       
      6 ♣
       
    </Segment>
    <Segment
      className="bid-sequence-cell DOUBLE-cell"
      key="1"
    >
       
      X
       
    </Segment>
    <Segment
      className="bid-sequence-cell REDOUBLE-cell"
      key="2"
    >
       
      XX
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="1"
  >
     
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="3"
    >
       
      Pass
       
    </Segment>
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="4"
    >
       
      Pass
       
    </Segment>
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="5"
    >
       
      Pass
       
    </Segment>
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: SOUTH, bidSeq:  1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell empty-cell"
      key="WEST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell empty-cell"
      key="NORTH"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell empty-cell"
      key="EAST"
    >
       Empty 
    </Segment>
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: SOUTH, bidSeq: [object Object],[object Object],[object Object],[object Object],[object Object] 1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell empty-cell"
      key="WEST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell empty-cell"
      key="NORTH"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell empty-cell"
      key="EAST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="green"
      inverted={true}
      key="0"
    >
       
      1 ♣
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="1"
  >
     
    <Segment
      className="bid-sequence-cell"
      color="orange"
      inverted={true}
      key="1"
    >
       
      2 ♦
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="red"
      inverted={true}
      key="2"
    >
       
      3 ♥
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="blue"
      inverted={true}
      key="3"
    >
       
      4 ♠
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="black"
      inverted={true}
      key="4"
    >
       
      5 N
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="2"
  >
     
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: SOUTH, bidSeq: [object Object],[object Object],[object Object],[object Object],[object Object],[object Object] 1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell empty-cell"
      key="WEST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell empty-cell"
      key="NORTH"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell empty-cell"
      key="EAST"
    >
       Empty 
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="green"
      inverted={true}
      key="0"
    >
       
      6 ♣
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="1"
  >
     
    <Segment
      className="bid-sequence-cell DOUBLE-cell"
      key="1"
    >
       
      X
       
    </Segment>
    <Segment
      className="bid-sequence-cell REDOUBLE-cell"
      key="2"
    >
       
      XX
       
    </Segment>
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="3"
    >
       
      Pass
       
    </Segment>
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="4"
    >
       
      Pass
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="2"
  >
     
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="5"
    >
       
      Pass
       
    </Segment>
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: WEST, bidSeq:  1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: WEST, bidSeq: [object Object],[object Object],[object Object],[object Object],[object Object] 1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell"
      color="green"
      inverted={true}
      key="0"
    >
       
      1 ♣
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="orange"
      inverted={true}
      key="1"
    >
       
      2 ♦
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="red"
      inverted={true}
      key="2"
    >
       
      3 ♥
       
    </Segment>
    <Segment
      className="bid-sequence-cell"
      color="blue"
      inverted={true}
      key="3"
    >
       
      4 ♠
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="1"
  >
     
    <Segment
      className="bid-sequence-cell"
      color="black"
      inverted={true}
      key="4"
    >
       
      5 N
       
    </Segment>
     
  </div>
</div>
`;

exports[`BidSequenceBlock render seat: WEST, bidSeq: [object Object],[object Object],[object Object],[object Object],[object Object],[object Object] 1`] = `
<div
  className="bid-sequence-block"
>
  <div
    className="bid-sequence-row"
    key="0"
  >
     
    <Segment
      className="bid-sequence-cell"
      color="green"
      inverted={true}
      key="0"
    >
       
      6 ♣
       
    </Segment>
    <Segment
      className="bid-sequence-cell DOUBLE-cell"
      key="1"
    >
       
      X
       
    </Segment>
    <Segment
      className="bid-sequence-cell REDOUBLE-cell"
      key="2"
    >
       
      XX
       
    </Segment>
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="3"
    >
       
      Pass
       
    </Segment>
     
  </div>
  <div
    className="bid-sequence-row"
    key="1"
  >
     
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="4"
    >
       
      Pass
       
    </Segment>
    <Segment
      className="bid-sequence-cell PASS-cell"
      key="5"
    >
       
      Pass
       
    </Segment>
     
  </div>
</div>
`;

exports[`BidSequenceDisplay render basic case 1`] = `
<div
  className="bid-sequence-display"
>
  <SeatRow
    vulnerability="EW"
  />
  <Divider />
  <BidSequenceBlock
    bidSeq={Array []}
    dealer="NORTH"
  />
</div>
`;

exports[`SeatRow render vulnerability: BOTH 1`] = `
<div
  className="seat-row"
>
  <Segment
    className="seat-cell"
    color="red"
    inverted={true}
    key="WEST"
  >
     
    WEST
     
  </Segment>
  <Segment
    className="seat-cell"
    color="red"
    inverted={true}
    key="NORTH"
  >
     
    NORTH
     
  </Segment>
  <Segment
    className="seat-cell"
    color="red"
    inverted={true}
    key="EAST"
  >
     
    EAST
     
  </Segment>
  <Segment
    className="seat-cell"
    color="red"
    inverted={true}
    key="SOUTH"
  >
     
    SOUTH
     
  </Segment>
</div>
`;

exports[`SeatRow render vulnerability: EW 1`] = `
<div
  className="seat-row"
>
  <Segment
    className="seat-cell"
    color="red"
    inverted={true}
    key="WEST"
  >
     
    WEST
     
  </Segment>
  <Segment
    className="seat-cell"
    key="NORTH"
  >
     
    NORTH
     
  </Segment>
  <Segment
    className="seat-cell"
    color="red"
    inverted={true}
    key="EAST"
  >
     
    EAST
     
  </Segment>
  <Segment
    className="seat-cell"
    key="SOUTH"
  >
     
    SOUTH
     
  </Segment>
</div>
`;

exports[`SeatRow render vulnerability: NONE 1`] = `
<div
  className="seat-row"
>
  <Segment
    className="seat-cell"
    key="WEST"
  >
     
    WEST
     
  </Segment>
  <Segment
    className="seat-cell"
    key="NORTH"
  >
     
    NORTH
     
  </Segment>
  <Segment
    className="seat-cell"
    key="EAST"
  >
     
    EAST
     
  </Segment>
  <Segment
    className="seat-cell"
    key="SOUTH"
  >
     
    SOUTH
     
  </Segment>
</div>
`;

exports[`SeatRow render vulnerability: NS 1`] = `
<div
  className="seat-row"
>
  <Segment
    className="seat-cell"
    key="WEST"
  >
     
    WEST
     
  </Segment>
  <Segment
    className="seat-cell"
    color="red"
    inverted={true}
    key="NORTH"
  >
     
    NORTH
     
  </Segment>
  <Segment
    className="seat-cell"
    key="EAST"
  >
     
    EAST
     
  </Segment>
  <Segment
    className="seat-cell"
    color="red"
    inverted={true}
    key="SOUTH"
  >
     
    SOUTH
     
  </Segment>
</div>
`;
